1.1.0b6 alpine linux build fails

Dockerfile:
```
FROM alpine:3.6

RUN apk add --no-cache bash python3 lz4-libs libattr libacl libressl ca-certificates \
        && pip3 install --upgrade pip \
        && apk add --no-cache --virtual .build-deps \
           gcc libc-dev make pcre-dev zlib-dev \
           python3-dev lz4-dev acl-dev attr-dev libressl-dev linux-headers\
        && pip3 install borgbackup==1.1.0b6 \
        && apk del .build-deps

```

Some error output:
```

...
    running build_ext
    building 'borg.compress' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/src
    creating build/temp.linux-x86_64-3.6/src/borg
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Os -fomit-frame-pointer -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -I/usr/include -I/usr/include -I/usr/include/python3.6m -c src/borg/compress.c -o build/temp.linux-x86_64-3.6/src/borg/compress.o
    src/borg/compress.c: In function '__pyx_pf_4borg_8compress_3LZ4_2compress':
    src/borg/compress.c:2517:9: warning: 'LZ4_compress_limitedOutput' is deprecated: use LZ4_compress_default() instead [-Wdeprecated-declarations]
             __pyx_v_osize = LZ4_compress_limitedOutput(__pyx_v_source, __pyx_v_dest, __pyx_v_isize, __pyx_v_osize);
             ^~~~~~~~~~~~~
    In file included from src/borg/compress.c:432:0:
    /usr/include/lz4.h:432:58: note: declared here
     LZ4_DEPRECATED("use LZ4_compress_default() instead") int LZ4_compress_limitedOutput (const char* source, char* dest, int sourceSize, int maxOutputSize);
                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
    gcc -shared -Wl,--as-needed -Wl,--as-needed build/temp.linux-x86_64-3.6/src/borg/compress.o -L/usr/lib -L/usr/lib -L/usr/lib -llz4 -lpython3.6m -o build/lib.linux-x86_64-3.6/borg/compress.cpython-36m-x86_64-linux-gnu.so
    building 'borg.crypto.low_level' extension
    creating build/temp.linux-x86_64-3.6/src/borg/crypto
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Os -fomit-frame-pointer -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -I/usr/include -I/usr/include -I/usr/include/python3.6m -c src/borg/crypto/low_level.c -o build/temp.linux-x86_64-3.6/src/borg/crypto/low_level.o
    gcc -shared -Wl,--as-needed -Wl,--as-needed build/temp.linux-x86_64-3.6/src/borg/crypto/low_level.o -L/usr/lib -L/usr/lib -L/usr/lib -lcrypto -lpython3.6m -o build/lib.linux-x86_64-3.6/borg/crypto/low_level.cpython-36m-x86_64-linux-gnu.so
    building 'borg.hashindex' extension
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Os -fomit-frame-pointer -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -I/usr/include/python3.6m -c src/borg/hashindex.c -o build/temp.linux-x86_64-3.6/src/borg/hashindex.o
    gcc -shared -Wl,--as-needed -Wl,--as-needed build/temp.linux-x86_64-3.6/src/borg/hashindex.o -L/usr/lib -lpython3.6m -o build/lib.linux-x86_64-3.6/borg/hashindex.cpython-36m-x86_64-linux-gnu.so
    building 'borg.item' extension
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Os -fomit-frame-pointer -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -I/usr/include/python3.6m -c src/borg/item.c -o build/temp.linux-x86_64-3.6/src/borg/item.o
    gcc -shared -Wl,--as-needed -Wl,--as-needed build/temp.linux-x86_64-3.6/src/borg/item.o -L/usr/lib -lpython3.6m -o build/lib.linux-x86_64-3.6/borg/item.cpython-36m-x86_64-linux-gnu.so
    building 'borg.chunker' extension
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Os -fomit-frame-pointer -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -I/usr/include/python3.6m -c src/borg/chunker.c -o build/temp.linux-x86_64-3.6/src/borg/chunker.o
    gcc -shared -Wl,--as-needed -Wl,--as-needed build/temp.linux-x86_64-3.6/src/borg/chunker.o -L/usr/lib -lpython3.6m -o build/lib.linux-x86_64-3.6/borg/chunker.cpython-36m-x86_64-linux-gnu.so
    building 'borg.algorithms.checksums' extension
    creating build/temp.linux-x86_64-3.6/src/borg/algorithms
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Os -fomit-frame-pointer -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -I/usr/include/python3.6m -c src/borg/algorithms/checksums.c -o build/temp.linux-x86_64-3.6/src/borg/algorithms/checksums.o
    In file included from src/borg/algorithms/crc32_dispatch.c:3:0,
                     from src/borg/algorithms/checksums.c:435:
    src/borg/algorithms/crc32_slice_by_8.c:40:2: error: #error Unknown byte order
     #error Unknown byte order
      ^~~~~
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ir8htrzr/borgbackup/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-d3cfom2r-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-ir8htrzr/borgbackup/
The command '/bin/sh -c apk add --no-cache bash python3 lz4-libs libattr libacl libressl ca-certificates         && pip3 install --upgrade pip         && apk add --no-cache --virtual .build-deps            gcc libc-dev make pcre-dev zlib-dev            python3-dev lz4-dev acl-dev attr-dev libressl-dev linux-headers        && pip3 install borgbackup==1.1.0b6         && apk del .build-deps' returned a non-zero code: 1

```

--- a/src/borg/_endian.h
+++ b/src/borg/_endian.h
@@ -1,0 +1,27 @@
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#if defined (__SVR4) && defined (__sun)
+#include <sys/isa_defs.h>
+#endif
+
+#if (defined(BYTE_ORDER) && defined(BIG_ENDIAN) && (BYTE_ORDER == BIG_ENDIAN)) ||  \
+    (defined(__BYTE_ORDER__) && defined(__ORDER_BIG_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)) || \
+    (defined(_BIG_ENDIAN) && defined(__SVR4) && defined(__sun))
+#define BORG_BIG_ENDIAN 1
+#elif (defined(BYTE_ORDER) && defined(LITTLE_ENDIAN) && (BYTE_ORDER == LITTLE_ENDIAN)) || \
+      (defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)) || \
+      (defined(_LITTLE_ENDIAN) && defined(__SVR4) && defined(__sun))
+#define BORG_BIG_ENDIAN 0
+#else
+#error Unknown byte order
+#endif
+
+#if BORG_BIG_ENDIAN
+#define _le32toh(x) __builtin_bswap32(x)
+#define _htole32(x) __builtin_bswap32(x)
+#else
+#define _le32toh(x) (x)
+#define _htole32(x) (x)
+#endif

--- a/src/borg/_hashindex.c
+++ b/src/borg/_hashindex.c
@@ -10,29 +10,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 
-#if defined (__SVR4) && defined (__sun)
-#include <sys/isa_defs.h>
-#endif
-
-#if (defined(BYTE_ORDER) && (BYTE_ORDER == BIG_ENDIAN)) ||  \
-    (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)) || \
-    (defined(_BIG_ENDIAN) && defined(__SVR4)&&defined(__sun))
-#define BORG_BIG_ENDIAN 1
-#elif (defined(BYTE_ORDER) && (BYTE_ORDER == LITTLE_ENDIAN)) || \
-      (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)) || \
-      (defined(_LITTLE_ENDIAN) && defined(__SVR4)&&defined(__sun))
-#define BORG_BIG_ENDIAN 0
-#else
-#error Unknown byte order
-#endif
-
-#if BORG_BIG_ENDIAN
-#define _le32toh(x) __builtin_bswap32(x)
-#define _htole32(x) __builtin_bswap32(x)
-#else
-#define _le32toh(x) (x)
-#define _htole32(x) (x)
-#endif
+#include "_endian.h"
 
 #define MAGIC "BORG_IDX"
 #define MAGIC_LEN 8

--- a/src/borg/algorithms/crc32_slice_by_8.c
+++ b/src/borg/algorithms/crc32_slice_by_8.c
@@ -9,6 +9,8 @@
 // size_t
 #include <stddef.h>
 
+#include "../_endian.h"
+
 /// compute CRC32 (Slicing-by-8 algorithm), unroll inner loop 4 times
 uint32_t crc32_4x8bytes(const void* data, size_t length, uint32_t previousCrc32);
 
@@ -22,23 +24,6 @@ uint32_t crc32_4x8bytes(const void* data, size_t length, uint32_t previousCrc32)
 
 /// zlib's CRC32 polynomial
 const uint32_t Polynomial = 0xEDB88320;
-
-/// swap endianness
-#if defined (__SVR4) && defined (__sun)
-#include <sys/isa_defs.h>
-#endif
-
-#if (defined(BYTE_ORDER)&&(BYTE_ORDER == BIG_ENDIAN)) ||  \
-    (defined(_BIG_ENDIAN)&&defined(__SVR4)&&defined(__sun))
-#define _le32toh(x) __builtin_bswap32(x)
-#define BORG_BIG_ENDIAN
-#elif (defined(BYTE_ORDER)&&(BYTE_ORDER == LITTLE_ENDIAN)) || \
-      (defined(_LITTLE_ENDIAN)&&defined(__SVR4)&&defined(__sun))
-#define _le32toh(x) (x)
-#define BORG_LITTLE_ENDIAN
-#else
-#error Unknown byte order
-#endif
 
 // //////////////////////////////////////////////////////////
 // constants
@@ -358,7 +343,7 @@ uint32_t crc32_slice_by_8(const void* data, size_t length, uint32_t previousCrc3
     size_t unrolling;
     for (unrolling = 0; unrolling < Unroll; unrolling++)
     {
-#ifdef BORG_BIG_ENDIAN
+#if BORG_BIG_ENDIAN
       uint32_t one = *current++ ^ _le32toh(crc);
       uint32_t two = *current++;
       crc = Crc32Lookup[0][ two      & 0xFF] ^

